src/containerapp-preview/azext_containerapp_preview/containerapp_decorator.py

```diff
@@ -0,0 +1,98 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from typing import Dict, Any
+from knack.log import get_logger
+
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.core.azclierror import ValidationError, RequiredArgumentMissingError
+from msrestazure.tools import parse_resource_id
+
+from ._clients import ManagedEnvironmentClient, ConnectedEnvironmentClient, ContainerAppClient
+from ._constants import GA_CONTAINERAPP_EXTENSION_NAME
+from ._utils import (_get_azext_module)
+
+logger = get_logger(__name__)
+
+
+def get_containerapp_base_decorator(cmd, raw_parameters):
+    azext_decorator = _get_azext_module("azext_containerapp.containerapp_decorator")
+
+    containerapp_base_decorator = azext_decorator.BaseContainerAppDecorator(
+        cmd=cmd,
+        client=ContainerAppClient,
+        raw_parameters=raw_parameters,
+        models="azext_containerapp._sdk_models"
+    )
+    return containerapp_base_decorator
+
+
+class ContainerAppPreviewCreateDecorator(_get_azext_module("azext_containerapp.containerapp_decorator").ContainerAppCreateDecorator):
+    def __init__(
+        self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str
+    ):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.azext_decorator_utils = _get_azext_module("azext_containerapp._decorator_utils")
+        self.azext_default_utils = _get_azext_module("azext_containerapp._utils")
+
+    def construct_containerapp(self):
+        containerapp_def = super().construct_containerapp()
+        containerapp_def = self.set_up_extended_location(containerapp_def)
+        return containerapp_def
+
+    def set_up_extended_location(self, containerapp_def):
+        if self.get_argument_environment_type() == "connected":
+            parsed_env = parse_resource_id(self.get_argument_managed_env())  # custom_location check here perhaps
+            env_name = parsed_env['name']
+            env_rg = parsed_env['resource_group']
+            env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+            if not containerapp_def.get('extendedLocation'):
+                containerapp_def["extendedLocation"] = env_info["extendedLocation"]
+        return containerapp_def
+
+    def get_environment_client(self):
+        if self.get_argument_yaml():
```

Comment:
no need to reload yaml value.

Reply:
TODO with refator

Reply:
[Task 24502107](https://msazure.visualstudio.com/Antares/_workitems/edit/24502107): ContainerAppCreateDecorator keep containerapp_def context